mantidproject/mantid
# In Vates subdirectory, replace iterators with range-based for loops.

Description of work.

While working on other issues, I found several places in the Vates subdirectory where a range-based for loop would be shorter and clearer than a for loop with iterators. For `Qlist` and `Qset` I used Qt's `foreach` macro to avoid potential [performance issues](http://www.dvratil.cz/2015/06/qt-containers-and-c11-range-based-loops/).

CompositePeaksPresenterVsi was already refactored in PR #15678. Those changes are NOT included here.

**To test:**



Check that I didn't significantly change the behavior of any of these member functions. 

This PR has no associated issue number.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
